### PR TITLE
Servlet 6.1 Add Test for New HTTP Response Status Codes

### LINKED ISSUE #26491: Servlet 6.1: Add new constants to HttpServletResponse for the HTTP response codes 308, 421, 422 and 426.
`https://github.com/jakartaee/servlet/issues/336`

### PR DESCRIPTION
**Pull request by pmd1nh**
fixes https://github.com/OpenLiberty/open-liberty/issues/26491

### CODE REVIEW

**Comment by pmd1nh** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.1.internal_fat

**Comment by pmd1nh** [comment]
io.openliberty.webcontainer.servlet.6.1.internal_fat FAT tests 2 0 0
	
io.openliberty.webcontainer.servlet61.fat.tests.Servlet61CharsetEncodingTest 1 0 0 1 s 1
io.openliberty.webcontainer.servlet61.fat.tests.Servlet61HTTPResponseCodesTest 1 0 0 1 s 2

File: dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/publish/servers/servlet61_HttpResponseCodesTest/bootstrap.properties [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License 2.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-2.0/
+# 
+# SPDX-License-Identifier: EPL-2.0
+###############################################################################
+bootstrap.include=../testports.properties
+osgi.console=7777
+com.ibm.ws.logging.trace.specification=*=info:com.ibm.ws.webcontainer*=all:com.ibm.wsspi.webcontainer*=all:GenericBNF=all:io.openliberty.webcontainer*=all
```
Comment (pnicolucci) [inline dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/publish/servers/servlet61_HttpResponseCodesTest/bootstrap.properties]: For a test of this nature, where we are really just testing some constants in the API is it really necessary to generate all  of this trace?

File: dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/publish/servers/servlet61_HttpResponseCodesTest/server.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<!--
+    Copyright (c) 2023 IBM Corporation and others.
+    All rights reserved. This program and the accompanying materials
+    are made available under the terms of the Eclipse Public License 2.0
+    which accompanies this distribution, and is available at
+    http://www.eclipse.org/legal/epl-2.0/
+    
+    SPDX-License-Identifier: EPL-2.0
+ -->
+<server description="Server for testing new optional HTTP response status code 308, 421, 422, 426 ">
```
Comment (pnicolucci) [inline dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/publish/servers/servlet61_HttpResponseCodesTest/server.xml]: What do you mean by "optional"?

File: dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/test-applications/HTTPResponseCodesTest.war/src/httpcodes/servlets/TestHttpResponseCodesServlet.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import jakarta.servlet.ServletContext;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletOutputStream;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+/**
+ * Test new optional HTTP response status code 308, 421, 422, 426
```
Comment (pnicolucci) [inline dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/test-applications/HTTPResponseCodesTest.war/src/httpcodes/servlets/TestHttpResponseCodesServlet.java]: What do you mean by optional?

File: dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/publish/servers/servlet61_HttpResponseCodesTest/server.xml (hunk 2 above)
Comment (pmd1nh) [reply dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/publish/servers/servlet61_HttpResponseCodesTest/server.xml]: they are not optional but new constant codes.  Updated.

File: dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/test-applications/HTTPResponseCodesTest.war/src/httpcodes/servlets/TestHttpResponseCodesServlet.java (hunk 3 above)
Comment (pmd1nh) [reply dev/io.openliberty.webcontainer.servlet.6.1.internal_fat/test-applications/HTTPResponseCodesTest.war/src/httpcodes/servlets/TestHttpResponseCodesServlet.java]: updated.

**Comment by pmd1nh** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.webcontainer.servlet.6.1.internal_fat

**Review by pnicolucci — APPROVED** [review]

**Comment by pmd1nh** [comment]
#libby

**Pull request merged**